static/js/publisher/form/banner.js

```diff
@@ -0,0 +1,217 @@
+    this.handleRemoveImageClick = this.handleRemoveImageClick.bind(this);
+
+    this.keyboardEvent = this.keyboardEvent.bind(this);
+    this.toggleFocus = this.toggleFocus.bind(this);
+
+    this.state = {
+      bannerImage: this.props.bannerImage,
+      focused: false,
+      errors: {}
+    };
+  }
+
+  bannerImageChangeHandler(files) {
+    const { updateImageState } = this.props;
+    const file = files[0];
+
+    if (file.errors) {
+      this.setState({
+        errors: {
+          [file.name]: file.errors
+        }
+      });
+    } else {
+      const url = URL.createObjectURL(file);
+
+      this.setState({
+        errors: {},
+        bannerImage: {
+          url: url
+        }
+      });
+
+      updateImageState({
+        url: url,
+        file,
+        name: file.name,
+        status: "new",
+        type: "banner"
+      });
+    }
+  }
+
+  handleRemoveImageClick(e) {
+    e.stopPropagation();
+    const { updateImageState } = this.props;
+
+    this.setState({
+      bannerImage: {}
+    });
+
+    updateImageState(null);
+  }
+
+  keyboardEvent(e) {
+    if (e.key === "Delete" || e.key === "Backspace") {
+      this.handleRemoveImageClick(e);
+    }
+  }
+
+  toggleFocus() {
+    const { focused } = this.state;
+    this.setState({
+      focused: !focused
+    });
+  }
+
+  renderErrors() {
+    const { errors } = this.state;
+    if (Object.keys(errors).length > 0) {
+      return (
+        <div className="p-notification--negative">
+          <p className="p-notification__response">
+            {Object.keys(errors).map(fileName => (
+              <Fragment key={`errors-${fileName}`}>
+                {fileName}
+                &nbsp;
+                {errors[fileName].map((error, index) => (
+                  <Fragment key={`errors-${fileName}-${index}`}>
+                    {error}
+                    <br />
+                  </Fragment>
+                ))}
+              </Fragment>
+            ))}
+          </p>
+        </div>
+      );
+    }
+    return false;
+  }
+
+  renderBackground() {
+    const { bannerImage } = this.state;
+    const { restrictions } = this.props;
+    const background = bannerImage.url;
+
+    let backgroundClasses = ["p-market-banner__image", "u-vertically-center"];
+    if (!background) {
+      backgroundClasses.push("is-empty");
+    }
+
+    return (
+      <Fragment>
+        <FileInput
+          inputName="banner-image"
+          inputId="banner-image"
+          active={true}
+          fileChangedCallback={this.bannerImageChangeHandler}
+          noFocus={true}
+          restrictions={restrictions}
+        >
+          <div
+            className={backgroundClasses.join(" ")}
+            tabIndex="0"
+            onKeyDown={this.keyboardEvent}
+            onFocus={this.toggleFocus}
+            onBlur={this.toggleFocus}
+          >
+            {background && <img src={background} />}
+            {!background && (
```

Comment:
Why not using `background ? <img src={background} /> : ( ... )` here?
It's not a nicest syntax, but at least it's closer to `if ... else` that it seems we want to achieve here.